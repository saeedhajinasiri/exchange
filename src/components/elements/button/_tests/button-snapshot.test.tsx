import React from "react";
import {cleanup, render} from "test";
import {Button, Text} from "@elements";

afterEach(cleanup)

describe('Test Button UI', () => {
  it('Default', () => {
    const {asFragment} = render(<Button>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with children prop', () => {
    const children = <Text color={"primary"}>children</Text>
    const {asFragment} = render(<Button>{children}</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with className prop', () => {
    const {asFragment} = render(<Button className={'test-class-name'}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Button Variants', () => {
  it('variant=text', () => {
    const {asFragment} = render(<Button variant={"text"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('variant=outlined', () => {
    const {asFragment} = render(<Button variant={"outlined"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('variant=contained', () => {
    const {asFragment} = render(<Button variant={"contained"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Button Colors', () => {
  it('color=primary', () => {
    const {asFragment} = render(<Button color={"primary"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=secondary', () => {
    const {asFragment} = render(<Button color={"secondary"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=tertiary', () => {
    const {asFragment} = render(<Button color={"tertiary"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=grey', () => {
    const {asFragment} = render(<Button color={"grey"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=light', () => {
    const {asFragment} = render(<Button color={"light"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=dark', () => {
    const {asFragment} = render(<Button color={"dark"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Button Sizes', () => {
  it('size=small', () => {
    const {asFragment} = render(<Button size={"small"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('size=medium', () => {
    const {asFragment} = render(<Button size={"medium"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('size=large', () => {
    const {asFragment} = render(<Button size={"large"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Button Shapes', () => {
  it('shape=circle', () => {
    const {asFragment} = render(<Button shape={"circle"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shape=square', () => {
    const {asFragment} = render(<Button shape={"square"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shape=rectangle', () => {
    const {asFragment} = render(<Button shape={"rectangle"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shape=circle', () => {
    const {asFragment} = render(<Button shape={"circle"}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Button Disabled', () => {
  it('disabled=true', () => {
    const {asFragment} = render(<Button disabled={true}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('disabled=false', () => {
    const {asFragment} = render(<Button disabled={false}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Button Loading', () => {
  it('loading=true', () => {
    const {asFragment} = render(<Button loading={true}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('loading=false', () => {
    const {asFragment} = render(<Button loading={false}>children</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
