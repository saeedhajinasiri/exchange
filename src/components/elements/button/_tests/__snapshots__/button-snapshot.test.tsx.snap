// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Colors color=dark 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-grey-dark border-grey text-grey hover:opacity-70 active:bg-grey-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Colors color=grey 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-grey border-grey text-grey hover:opacity-70 active:bg-grey-dark active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Colors color=light 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-white border-grey text-grey hover:opacity-70 active:bg-grey-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Colors color=primary 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Colors color=secondary 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-secondary border-secondary text-secondary hover:opacity-70 active:bg-secondary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Colors color=tertiary 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-tertiary border-tertiary text-tertiary hover:opacity-70 active:bg-tertiary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Disabled disabled=false 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Disabled disabled=true 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out h-12 text-white text-m-sm md:text-d-sm border-none bg-grey-light rounded-md px-8"
    data-testid="button-test-id"
    disabled=""
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Loading loading=false 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Loading loading=true 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    <svg
      class="animate-spin h-5 w-5 text-white"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="opacity-25"
        cx="12"
        cy="12"
        r="10"
        stroke="grey"
        stroke-width="4"
      />
      <path
        class="opacity-75"
        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        fill="currentColor"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Test Button Shapes shape=circle 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md max-w-[48px] min-w-[48px] w-12 px-0rounded-full"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Shapes shape=circle 2`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md max-w-[48px] min-w-[48px] w-12 px-0rounded-full"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Shapes shape=rectangle 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Shapes shape=square 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md max-w-[48px] min-w-[48px] w-12 px-0"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Sizes size=large 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-14 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Sizes size=medium 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Sizes size=small 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-9 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button UI Default 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button UI Test with children prop 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
    >
      children
    </p>
  </button>
</DocumentFragment>
`;

exports[`Test Button UI Test with className prop 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8 test-class-name"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Variants variant=contained 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 text-white text-m-sm border-none md:text-d-sm rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Variants variant=outlined 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 border bg-transparent rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;

exports[`Test Button Variants variant=text 1`] = `
<DocumentFragment>
  <button
    class="flex items-center justify-center hover:transition hover:duration-150 hover:ease-in-out bg-primary border-primary text-primary hover:opacity-70 active:bg-primary-light active:text-white h-12 bg-transparent border-none rounded-md px-8"
    data-testid="button-test-id"
  >
    children
  </button>
</DocumentFragment>
`;
